File: android/app/libs/ChipDnaMobileJavaDemo/ChipDnaMobileDemoForAndroid/src/main/java/com/creditcall/chipdnamobiledemo/DigitalSignatureCaptureActivity.java
package com.creditcall.chipdnamobiledemo;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.os.Parcel;

import androidx.appcompat.app.AppCompatActivity;

import com.creditcall.chipdnamobile.ChipDnaMobileException;
import com.creditcall.chipdnamobile.ChipDnaMobileUtils;

@SuppressLint("ParcelCreator")
public class DigitalSignatureCaptureActivity extends AppCompatActivity implements SignatureCaptureFragment.SignatureCaptureInteractionListener {

	static final int DIGITAL_SIGNATURE_CAPTURE_ACTIVITY_CODE = 3;
	static final int SIGNATURE_CAPTURE_REQUEST_ACCEPTED = 1;
	static final int SIGNATURE_CAPTURE_REQUEST_TERMINATED = 2;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_digital_signature_capture);

		getSupportFragmentManager().beginTransaction()
				.add(R.id.frameLayout, SignatureCaptureFragment.newInstance(this)).commit();
	}

	@Override
	public void onTerminate() {
		setResult(SIGNATURE_CAPTURE_REQUEST_TERMINATED);
		finish();
	}

	@Override
	public void onSignatureAvailable(Bitmap signatureBitmap) {
		Intent intent = new Intent();
		try {
			intent.putExtra(DigitalSignatureVerificationActivity.SIGNATURE_DATA, ChipDnaMobileUtils.getSignaturePngStringFromBitmap(signatureBitmap));
			setResult(SIGNATURE_CAPTURE_REQUEST_ACCEPTED, intent);
		} catch (ChipDnaMobileException e) {
			// only thrown when Bitmap is not in PNG
			// the transaction could be accepted however this implementation does not
			e.printStackTrace();
			setResult(SIGNATURE_CAPTURE_REQUEST_TERMINATED, intent);
		}
		finish();
	}

	@Override
	public int describeContents() {
		return 0;
	}

	@Override
	public void writeToParcel(Parcel parcel, int i) {

	}
}

File: android/app/libs/ChipDnaMobileKotlinDemo/app/src/main/kotlin/com/creditcall/chipdnamobiledemo/ui/DigitalSignatureCaptureActivity.kt
package com.creditcall.chipdnamobiledemo.ui

import android.content.Intent
import android.graphics.Bitmap
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.creditcall.chipdnamobile.ChipDnaMobileException
import com.creditcall.chipdnamobile.ChipDnaMobileUtils
import com.creditcall.chipdnamobiledemo.R
import kotlinx.parcelize.Parcelize

@Parcelize
class DigitalSignatureCaptureActivity : AppCompatActivity(),
  SignatureCaptureFragment.SignatureCaptureInteractionListener {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_digital_signature_capture)
    supportFragmentManager.beginTransaction().add(
      R.id.frameLayout,
      SignatureCaptureFragment.newInstance(this)
    ).commit()
  }

  override fun onTerminate() {
    setResult(SIGNATURE_CAPTURE_REQUEST_TERMINATED)
    finish()
  }

  override fun onSignatureAvailable(signatureBitmap: Bitmap) {
    try {
      val intent = Intent().apply {
        putExtra(
          KEY_SIGNATURE_DATA,
          ChipDnaMobileUtils.getSignaturePngStringFromBitmap(signatureBitmap)
        )
      }
      setResult(SIGNATURE_CAPTURE_REQUEST_ACCEPTED, intent)
      finish()
    } catch (e: ChipDnaMobileException) {
      e.printStackTrace()
    }
  }

  companion object {
    val SIGNATURE_CAPTURE_REQUEST_ACCEPTED = 1
    val SIGNATURE_CAPTURE_REQUEST_TERMINATED = 2
    val KEY_SIGNATURE_DATA = "KEY_SIGNATURE_DATA"
  }

}